app/code/Magento/Quote/Model/Cart/AddProductsToCartInterface.php

```diff
@@ -0,0 +1,28 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Quote\Model\Cart;
+
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Quote\Model\Cart\Data\AddProductsToCartOutput;
+
+/**
+ * Unified approach to add products to the Shopping Cart.
```

Comment:
Should we introduce storefront API instead (in `Magento/Quote/Api` directory)? Then it should also be responsible for authorization checks. Current approach seems unreliable and can lead to pretty serious security vulnerabilities in case of mistake in the client implementation.